added documenation on new gpu resources (#273)
leap-stc/leap-stc.github.io: docs/compute/compute_scaling.md

### GPU

Selecting the **GPU** profile starts your container on a node with a **single** GPU attached. Multi-GPU servers are not offered on the hub. If you need data-parallel training across several GPUs, see [Batch Jobs](https://leap-stc.github.io/compute/skypilot/).

> Three GPU types are available from the **GPU type** dropdown:

Reviewer: ```suggestion
Three GPU types are available:
```

| GPU type            | VRAM  | RAM (limit)    | CPUs | Google Cloud node | Who can select it                       |
| ------------------- | ----- | -------------- | ---- | ----------------- | --------------------------------------- |
| **NVIDIA Tesla T4** | 16 GB | ~30 GB (29 GB) | 8    | `n1-standard-8`   | Anyone with GPU access (default choice) |
| **NVIDIA L4**       | 24 GB | ~16 GB (15 GB) | 4    | `g2-standard-4`   | Members of `leap-stc:large-gpu`         |
| **NVIDIA A100**     | 40 GB | ~85 GB (82 GB) | 12   | `a2-highgpu-1g`   | Members of `leap-stc:large-gpu`         |

> The first RAM figure matches the launcher dropdown (the node's total). The figure in parentheses is the hard limit on your server, so budget against that one.

Reviewer: Is the first RAM figure the VRAM or RAM (limit) column name?

Author: the first figure is the VRAM, so 24 is VRAM and 16 is RAM (limit)

!!! note "A bigger GPU isn't always faster"

>     Start on the T4 and move up only when you hit a specific wall: you run out of GPU memory, or you need something the T4 can't do. The L4 comes with a smaller machine around it, so choosing it just to get "more GPU" can leave you slower than where you started.

Reviewer: ```suggestion
    Start on the T4 and move up only when you hit a specific wall like running out of GPU memory. The L4 comes with a smaller machine around it, so choosing it just to get "more GPU" can leave you slower than where you started.
```

#### Data loading is often the bottleneck

Much LEAP work trains comparatively modest models on very large datasets read from cloud object storage. The GPU is frequently *not* the constraint: the CPUs feeding it are. LEAP's own [ClimSim](https://github.com/leap-stc/ClimSim) tooling carries this warning in its data utilities: used as a training dataloader, I/O can slow training down considerably.

Practical implications:

- **Profile before upgrading.** If `nvidia-smi` shows low GPU utilization during training, a bigger GPU will not help. More CPUs, larger chunk-aligned reads, or caching a subset to local disk will.
- **Watch the L4's 4 CPUs.** Multi-worker PyTorch `DataLoader` setups and Zarr decompression are CPU-bound. Four cores and a ~15 GB memory ceiling can leave a 24 GB GPU idle.

> - **Read along your chunk boundaries.** Random access across a Zarr store's chunk grid amplifies reads and starves the GPU regardless of which one you picked.

Reviewer 2: Could you add an example here?

- **Consider CPU-only for offline evaluation.** Scoring predictions, computing metrics, and plotting rarely benefit from a GPU.

#### Requesting access to the L4 and A100

> Both large GPUs are gated behind a GitHub team so these scarce, expensive nodes stay available to the researchers who need them.

Reviewer 2: "who need them" -> "whose tasks can't be done with less expensive resources" 

(just so we aren't implying anything about what people do and don't "need")

1. Ask the LEAP Data and Compute Team (see [Contact](https://leap-stc.github.io/support/contact/)) to add your GitHub username to **`leap-stc:large-gpu`**. Include a short description of your workflow and why the T4 is not sufficient.
1. Accept the GitHub team invitation.
1. **Log out of the hub and log back in.** Your team membership is read at login, so the new options will not appear until you re-authenticate.

#### Availability and startup time

All GPU node pools run in **us-central1** and scale down to zero when unused.

- **Expect a slow first launch.** If no GPU node of your chosen type is already running, one must be provisioned before your server starts. This commonly takes several minutes.
- **Capacity is not guaranteed.** Google Cloud GPU availability varies by zone and week. The **L4 has been persistently scarce**; the A100 is generally easier to obtain.
- **If your server never starts**, try a different GPU type, or wait and retry later. If it fails for more than a few hours, [contact us](https://leap-stc.github.io/support/contact/).

#### Checking that your GPU is working

> From a terminal on your server:

Reviewer: Might be worth double checking from a fresh jupyter-lab session that these commands work out of the box.

#### Why not use GPU by default?

While GPU can accelerate certain workloads, it's not always the best choice:

- **Most tasks don't benefit:** Plotting, pandas/xarray analysis, and basic modeling run just as fast (or faster) on CPUs.
- **Shared, limited resources:** GPUs are a shared resource across LEAP. Using them when not needed can block others who rely on them for large-scale work.
- **More costly:** GPUs cost significantly more than CPU resources, and the A100 is the most expensive option on the hub by a wide margin.

!!! note

    Please use GPUs only if your workflow truly needs it, and **stop your server as soon as you are done**, an idle GPU server bills at the full GPU rate.

!!! tip "Running a workshop or class that needs GPUs?"

    A large GPU can be time-sliced across users. Time-slicing interleaves work rather than partitioning it: everyone shares one pool of VRAM with no memory or fault isolation, so one participant can OOM the others. Good for teaching, poor for unattended training. Not enabled by default.

## How to choose:

Here is a simplified guide on how to choose the appropriate image and compute configuration:

> | Scenario                                                   | Image                | Hardware                                          |

Reviewer: These examples are really nice @sarika-debruyn.

Maybe they could be trimmed down a tad? or split up a bit into more general workloads / tasks? ie, things that might need a GPU:
- Then 1-2 examples. ie, you're using pytorch or tensorflow.
